Fix #10687 - Product image roles disappearing

Scarraban opened the pull request:
To fix #10687 - adds the 'types' key to the images array to provide existing image roles, which are reset after clearMediaAttribute() is called.

### Description
This PR is to fix the issue outlined in #10687 - image roles being removed when saving by the ProductRepository.

### Fixed Issues (if relevant)
1. magento/magento2#10687: Product image roles randomly disappear

### Manual testing scenarios
1. Initially save a product using the ProductRepostory, setting an image in the gallery with roles.
2. Save the product again, without setting the media gallery at all. Roles should persist after second save.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=15606) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=15606) before we can accept your contribution.<br/>

Scarraban committed:
Fix #10687 - add types to images array

To fix #10687 - adds the 'types' key to the $images array to provide existing image roles (if not set), which are re-attributed after clearMediaAttribute() is called

magento-engcom-team commented:
Hi @sidolov, thank you for the review.
[ENGCOM-2672](https://jira.corp.magento.com/browse/ENGCOM-2672) has been created to process this Pull Request

magento-engcom-team commented:
@Scarraban thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

magento-engcom-team commented:
Hi @Scarraban. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.7. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

Scarraban commented:
Hi @michel334 ,

Check my replications steps comment on the issue #10687 - this was a patch fix that references the ProductRepository.php file before I made the pull request.
